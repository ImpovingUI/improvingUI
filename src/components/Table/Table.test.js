import { render, screen } from '@testing-library/react';
import { Table } from './'
const ejemplo = require('./Ejemplo.json');

describe('General table tests', () => {


  // test('Component is rendered', () => {
  //   render(<Table
  //     listColumns={["Name"]}
  //     listRows={[
  //       { Name: "Adrian" },
  //       { Name: "Garcia" },
  //       { Name: "Saaib" }
  //     ]}
  //   />)
  //   const component = screen.getByRole('data')
  //   component
  //   expect(component.container).toBeDefined(1)
  // });

  xtest('Component', () => {
    render(<Table
      listColumns={["Name", "Last Name", "City", "State", "Phone"]}
      listRows={ejemplo.datos}
    />)

    const tableHead = screen.queryAllByRole('row')
    expect(tableHead).toHaveLength(15)
  });

  xtest('Tabla exists', () => {
    const props = {
      listColumns: ["Name", "Last Name", "City", "State", "Phone"],
      listRows: ejemplo.datos,
      filter: true,
      pagination: true,
      listIndex: [3, 4, 5],
      actions: <div>Hola</div>,
      minInput: 1,
    }
    const component = render(<Table {...props} />)

    expect(component.container).toMatchSnapshot()
  });

  xtest('Tabla exists role', () => {
    const props = {
      listColumns: ["Name", "Last Name", "City", "State", "Phone"],
      listRows: ejemplo.datos,
    }
    render(<Table {...props} />)
    expect(screen.getByRole('data')).toBeTruthy()
  });
})
