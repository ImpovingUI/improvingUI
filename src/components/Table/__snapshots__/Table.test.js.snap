// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General table tests Tabla exists 1`] = `
<div>
  <div
    class="ContainerTable"
    role="data"
  >
    <div>
      <div>
        <input
          class="input-Table"
          type="text"
          value=""
        />
        <select
          class="select-Table"
        >
          <option
            value="all"
          >
            All
          </option>
          <option
            value="nombre"
          >
            Name
          </option>
          <option
            value="apellido"
          >
            Last Name
          </option>
          <option
            value="ciudad"
          >
            City
          </option>
          <option
            value="estado"
          >
            State
          </option>
          <option
            value="numero"
          >
            Phone
          </option>
        </select>
      </div>
    </div>
    <table
      class="Table"
    >
      <thead
        class="Thead-Table"
      >
        <tr>
          <th>
            Name
          </th>
          <th>
            Last Name
          </th>
          <th>
            City
          </th>
          <th>
            State
          </th>
          <th>
            Phone
          </th>
        </tr>
      </thead>
      <tbody
        class="Tbody-Table"
      >
        <tr>
          <td
            class="Td-Table message"
            colspan="5"
          >
            Type to search
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="containerPagin-Table"
    >
      <div>
        <select
          class="selectPage-Table"
        >
          <option
            value="5"
          >
            Items per page: 5
          </option>
          <option
            value="10"
          >
            Items per page: 10
          </option>
          <option
            value="50"
          >
            Items per page: 50
          </option>
          <option
            value="100"
          >
            Items per page: 100
          </option>
        </select>
      </div>
      <div>
        1
        -
        0
         of 
        0
         items
      </div>
      <div>
        1
         of 
        1
         pages
      </div>
      <div>
        <button
          class="btn-direction-Table"
        >
          ◄
        </button>
        <button
          class="btn-direction-Table"
        >
          ►
        </button>
      </div>
    </div>
  </div>
</div>
`;
